greenheart/tools/eco/hybrid_system.py

```diff
@@ -0,0 +1,456 @@
+
+    # this approach exactly sizes the energy needed for the non-electrolyzer components (according to the current models anyway)
+    # solver_result = optimize.brentq(energy_residual_function, -10, 20000, rtol=1E-5)
+    # OptimizeResult = optimize.root(energy_residual_function, 11E3, tol=1)
+    # solver_result = OptimizeResult.x
+    # solver_results = simple_solver(solver_result)
+    # solver_result = solver_results[0]
+    # print(solver_result)
+
+    ##################################################################################################################
+
+    # get results for final design
+    electrolyzer_physics_results, electrolyzer_cost_results, desal_results, h2_pipe_array_results, h2_transport_compressor, h2_transport_compressor_results, h2_transport_pipe_results, pipe_storage, h2_storage_results, total_accessory_power_renewable_kw, total_accessory_power_grid_kw \
+        = energy_internals(solver=False, power_for_peripherals_kw_in=solver_result)
+    
+    ## end solver loop here
+    platform_results = he_h2.run_equipment_platform(hopp_config, eco_config, orbit_config, design_scenario, hopp_results, electrolyzer_physics_results, h2_storage_results, desal_results, verbose=verbose)
+    
+    ################# OSW intermediate calculations" aka final financial calculations
+    # does LCOE even make sense if we are only selling the H2? I think in this case LCOE should not be used, rather LCOH should be used. Or, we could use LCOE based on the electricity actually used for h2
+    # I think LCOE is just being used to estimate the cost of the electricity used, but in this case we should just use the cost of the electricity generating plant since we are not selling to the grid. We
+    # could build in a grid connection later such that we use LCOE for any purchased electricity and sell any excess electricity after H2 production
+    # actually, I think this is what OSW is doing for LCOH
+    
+    # TODO double check full-system CAPEX
+    capex, capex_breakdown = he_fin.run_capex(hopp_results, orbit_project, orbit_hybrid_electrical_export_project, electrolyzer_cost_results, h2_pipe_array_results, h2_transport_compressor_results, h2_transport_pipe_results, h2_storage_results, hopp_config, eco_config, orbit_config, design_scenario, desal_results, platform_results, verbose=verbose)
+
+    # TODO double check full-system OPEX
+    opex_annual, opex_breakdown_annual = he_fin.run_opex(hopp_results, orbit_project, orbit_hybrid_electrical_export_project, electrolyzer_cost_results, h2_pipe_array_results, h2_transport_compressor_results, h2_transport_pipe_results, h2_storage_results, hopp_config, eco_config, orbit_config, desal_results, platform_results, verbose=verbose, total_export_system_cost=capex_breakdown["electrical_export_system"])
+
+    if verbose:
+        print("hybrid plant capacity factor: ", np.sum(hopp_results["combined_hybrid_power_production_hopp"])*1E-3/(eco_config["plant"]["capacity"]*365*24))
+
```

Comment:
Wouldn't it be better to use the power post-solver results (`remaining_power_profile`)?

Reply:
I corrected the scalar and the denominator here, but I don't think the hope_results portion of the numerator need to change

Reply:
Just to clarify, this capacity factor is for electricity, not hydrogen. If we use the `remaining_power_profile` we will not be accounting for all the electricity produced and used in the plant.